src/orquestra/sdk/_base/_workflow.py

```diff
@@ -256,6 +259,13 @@ def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> WorkflowDef[_R]:
             orquestra.sdk.exceptions.WorkflowSyntaxError: if the arguments don't match
                 the workflow def's parameters.
         """
+        allowed_function_calls = (
+            Secret,
+            secrets.get,
+            secrets.delete,
+            secrets.list,
+            secrets.set,
```

Comment:
**question**: is there any reason to call `secrets.{delete,list,set}` inside a workflow fn?

Reply:
No, but it seemed weird to me to allow getting secrets and none of the other bits of the API

Reply:
Now, if I use `secrets.list()` it would produce a submit-time constant, which is very different to `secrets.get()` which produces a Secret node.

Reply:
This wouldn't embed any secret values though.

This change here only related to if the `NotATaskWarning` is printed.

Reply:
Fair enough!